src/procedures/proc_pagerank.c

```diff
@@ -58,62 +78,105 @@ ProcedureResult Proc_PagerankInvoke(ProcedureCtx *ctx, const SIValue *args) {
 	ctx->privateData = pdata;
 
 	// Get label matrix.
-	s = GraphContext_GetSchema(gc, label, SCHEMA_NODE);
-	if(!s) return PROCEDURE_OK;
-	l = Graph_GetLabelMatrix(g, s->id);
+	if(label) {
+		s = GraphContext_GetSchema(gc, label, SCHEMA_NODE);
+		// Unknown label, quickly return.
+		if(!s) return PROCEDURE_OK;
+		l = Graph_GetLabelMatrix(g, s->id);
```

Comment:
I feel like it makes more sense to perform this work in the `if(label)` condition at 114.

Reply:
I prefer this notion of a "barrier", from this point onward we have both a label and relation matrices, which the next scope relies on.